src/sentry/api/endpoints/external_team_details.py

```diff
@@ -0,0 +1,62 @@
+import logging
+from django.http import Http404
+
+from rest_framework import status
+from rest_framework.response import Response
+
+from sentry.api.bases.team import TeamEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import ExternalTeam
+
+from .external_team import ExternalTeamSerializer
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalTeamDetailsEndpoint(TeamEndpoint):
+    def convert_args(
+        self, request, organization_slug, team_slug, external_team_id, *args, **kwargs
+    ):
+        args, kwargs = super().convert_args(request, organization_slug, team_slug, *args, **kwargs)
+        try:
+            kwargs["external_team"] = ExternalTeam.objects.get(
+                id=external_team_id,
+            )
+        except ExternalTeam.DoesNotExist:
+            raise Http404
+
+        return (args, kwargs)
+
+    def put(self, request, team, external_team):
```

Comment:
@NisanthanNanthakumar If the `POST` endpoint can do updating, do we really need the `PUT` as well? Might be unnecessary.

Reply:
@scefali hmm the POST endpoint cannot find by `external_team_id` and cannot update partially. I think we need both POST and PUT endpoints.

Reply:
@NisanthanNanthakumar I'm confused. If the `POST` endpoint is idempotent and we pass in the `external_id` as a parameter on the body, why do we need a `PUT`? The `POST` does a full update on the object, yea?

Reply:
@NisanthanNanthakumar Nvm, I see what you mean. We _want_ to do a partial update with the PUT. Not sure why we'd want to do a partial update but I see now how the endpoints can do different things.